build/repo.targets

```diff
@@ -7,4 +7,27 @@
       <AdditionalProperties>Configuration=$(Configuration)NoVSIX</AdditionalProperties>
     </Solutions>
   </ItemGroup>
+
+  <PropertyGroup>
+    <PrepareDependsOn>$(PrepareDependsOn);GenerateMSBuildLocationFile</PrepareDependsOn>
+    <RazorDesignTestProject>$(RepositoryRoot)test\Microsoft.AspNetCore.Razor.Design.Test\</RazorDesignTestProject>
+    <MSBuildLocationFileTemplate>$(RazorDesignTestProject)BuildVariables.cs.template</MSBuildLocationFileTemplate>
+    <MSBuildLocationFileOutput>$(RazorDesignTestProject)obj\BuildVariables.generated.cs</MSBuildLocationFileOutput>
+  </PropertyGroup>
+
+  <Target Name="GenerateMSBuildLocationFile"
+      DependsOnTargets="GetToolsets"
+      Inputs="$(MSBuildLocationFileTemplate);$(VisualStudioMSBuildx86Path)"
+      Outputs="$(MSBuildLocationFileOutput)">
+
+    <PropertyGroup>
+      <TemplateProperties>MSBuildLocation=$(VisualStudioMSBuildx86Path)</TemplateProperties>
+    </PropertyGroup>
+
+    <GenerateFileFromTemplate
```

Comment:
FYI this target is also available in a project context. Sdk_GenerateFileFromTemplate. You might consider making this part of the normal project build so VS scenarios work. You could detect the path to MSBuild.exe when building in VS by looking at build-in MSBuild properties such as MSBuildToolsPath.